### PR TITLE
feat: add delete phone endpoint to v2 api

### PR DESCRIPTION
**Pull request by jbiscoglia-klaviyo**
# Which Problems Are Solved

- Adds delete phone endpoint to v2 api

# How the Problems Are Solved

- Adds new endpoint with DELETE method to /v2beta/users/:userId/phone which removes currently set phone number

# Additional Changes

- Added integration test for new endpoint.

# Additional Context

- Solves https://discord.com/channels/927474939156643850/1255557862286032996

### CODE REVIEW

**Review by stebenz — CHANGES_REQUESTED** [review]

File: internal/api/grpc/user/v2/phone_integration_test.go [hunk 1]
```diff
@@ -245,3 +245,38 @@ func TestServer_VerifyPhone(t *testing.T) {
...
+
+func TestServer_RemovePhone(t *testing.T) {
+	userResp := Tester.CreateHumanUser(CTX)
+	tests := []struct {
+		name    string
+		req     *user.RemovePhoneRequest
+		want    *user.RemovePhoneResponse
+		wantErr bool
+	}{
+		{
```
[the remark this answers is not in the document]
Comment (jbiscoglia-klaviyo) [reply internal/api/grpc/user/v2/phone_integration_test.go:265]: Added those tests plus one for a missing user id. I'm throwing a phone not found error if the user has either already removed their phone or has never set one. Let me know if I should do something different.

[the remark this answers is not in the document]
Comment (jbiscoglia-klaviyo) [reply internal/api/grpc/user/v2/phone_integration_test.go:266]: Good catch. Should be good now

[the remark this answers is not in the document]
Comment (stebenz) [reply internal/api/grpc/user/v2/phone_integration_test.go:266]: Technically it works like this, but I would prefer it that you have the dependencies in the tests and no condition dependent on a string.
Just as an example, have a look in the [query_integration_test.go](https://github.com/zitadel/zitadel/blob/0ea3c5691f420fbb9644a9b2c7e5fc65aadebd3b/internal/api/grpc/user/v2/query_integration_test.go#L26).

[the remark this answers is not in the document]
Comment (jbiscoglia-klaviyo) [reply internal/api/grpc/user/v2/phone_integration_test.go:267]: Should be good now

**Review by eliobischof — APPROVED** [review]
Hi @jbiscoglia-klaviyo 
Thanks a lot for this helpful feature, we really appreciate it.
If you would like to get a small gift from us, please reach out to hi@zitadel.com

**Comment by eliobischof** [comment]
@jbiscoglia-klaviyo can you please merge back main or allow me to push to the PR?

**Comment by jbiscoglia-klaviyo** [comment]
Not sure why the e2e test failed. I think it might be a flaky test and I made no changes to the console. @eliobischof Are you able to manually re-run the workflow?

**Pull request merged**